frontend/src/components/EditableSelect.tsx

```diff
@@ -0,0 +1,151 @@
+import classNames from 'classnames';
+import {
+  FC,
+  useState,
+  ComponentPropsWithoutRef,
+  Dispatch,
+  SetStateAction,
+  useEffect,
+} from 'react';
+import Select from 'react-select';
+import { CloseButton, ConfirmButton, EditButton } from './forms/SvgButton';
+import { LoadingSpinner } from './LoadingSpinner';
+import { FieldError } from './forms/FormField';
+import css from './EditableSelect.module.scss';
+
+const classes = classNames.bind(css);
+
+interface Option {
+  value: number;
+  label: string;
+}
+
+interface WithButtonsProps {
+  onOk: (newValue: number, fieldId: string) => Promise<string | void>;
+  value: Option;
+  fieldId: string;
+  format: string | undefined;
+  options: Option[];
+}
+
+const withButtons = (Component: typeof EditableSelect) => ({
+  onOk,
+  value,
+  fieldId,
+  format,
+  options,
+}: WithButtonsProps) => {
+  const [editOpen, setEditOpen] = useState(false);
+  const [editValue, setEditValue] = useState<Option | undefined>(undefined);
```

Comment:
Would the edit value ever need to be `undefined`, or could it be initialised with `value`?